File: src/Services/SearchService/Domain/Entities/Auction.cs
using MongoDB.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities
{
    public class Auction : Entity
    {
        public int ReservePrice { get; set; }
        public string? Seller { get; set; }
        public string? Winner { get; set; }
        public int SoldAmount { get; set; }
        public int CurrentHighBid { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdateAt { get; set; }
        public DateTime AuctionEnd { get; set; }
        public string? Status { get; set; }
        public required Item Item { get; set; }
    }
}

File: frontend/web-app/types/Auction.ts
import { Item } from "./Item"

export type Auction = {
  id: string
  reservePrice: number
  seller: string
  winner?: string
  soldAmount: number
  currentHighBid: number
  createdAt: string
  updateAt: string
  auctionEnd: string
  status: string
  item: Item
}